codex/rest/api.nim

```diff
@@ -636,6 +636,35 @@ proc initDebugApi(node: CodexNodeRef, conf: CodexConf, router: var RestRouter) =
         trace "Excepting processing request", exc = exc.msg
         return RestApiResponse.error(Http500)
 
+  when codex_enable_api_debug_delete:
+    type
+      DebugDeleteParams = object
+        blockIndices* {.serialize.}: seq[int]
+
+    router.rawApi(
+      MethodPost,
+      "/api/codex/v1/debug/delete/{cid}") do (cid: Cid) -> RestApiResponse:
+        try:
+          without cid =? cid.tryGet.catch, error:
+            return RestApiResponse.error(Http400, error.msg)
+
+          let body = await request.getBody()
+          without params =? DebugDeleteParams.fromJson(body), error:
+            return RestApiResponse.error(Http400, error.msg)
+
+          warn "debug/delete is deleting blocks", cid = $cid, nBlocks = params.blockIndices.len
+
+          without deleted =? (await node.debugDelete(cid, params.blockIndices)), err:
+            return RestApiResponse.error(Http500, err.msg)
```

Comment:
You'll get a 500 if you supply an invalid block index which is just a bad request. I think this is OK for a debug API but not in general.

Reply:
Yeah, `400` is more appropriate.

Reply:
400 it is!

Reply:
The error codes should make it easier for API users to understand what they are doing wrong - so 400 for bad parameters (i.e., debug your parameters!) or 500 for actual errors (i.e., debug the code!). We're now mixing validation errors with actual failure to delete stuff, which could lead someone to waste time tweaking parameters when there's an actual bug in code. At that point I'd rather have the 500.

Reply:
You're right. node.debugDelete's failure should be a 500 in some cases and 400 in others. However, I think it's not really worth it to make this code any more complicated to achieve this. Instead, I added two error log statements to debugDelete. I hope that's an acceptable compromise.